foreman/data_refinery_foreman/surveyor/external_source.py

```diff
@@ -72,8 +43,24 @@ def determine_pipeline(self,
         """
         return
 
+    def downloader_task(self):
+        """Returns the downloader task for the source.
+
+        Returns the Celery Downloader Task name from the
+        data_refinery_workers project which should be queued to
+        download Batches discovered by this surveyor.
+        """
+        return DOWNLOADER_TASK_LOOKUP[self.source_type()]
+
     def handle_batches(self, batches: List[Batch]):
+        new_batches = []
         for batch in batches:
+            if not Batch.is_new_batch(batch):
```

Comment:
Mhmm, yeah, I see how that makes sense. I could also argue that `Batch.objects.filter(name=batch.name).count() > 0` is a standard Django way of doing things - people who use Django will use statements like that a lot to check if there is already an object in the table with those characteristics. However, what you say, in the context of Martin Fowler's blog post, makes sense, so I think this form also makes sense.